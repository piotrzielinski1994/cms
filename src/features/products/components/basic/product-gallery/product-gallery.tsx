import { Image } from '@/components/basic/image';
import { cn } from '@/utils/tailwind';
import { useId, useState } from 'react';

type ProductGalleryProps = {
  images: {
    src: string;
    alt: string;
  }[];
  className?: string;
};

const ProductGallery = ({ images, className, ...props }: ProductGalleryProps) => {
  const id = useId();
  const [activeIndex, setActiveIndex] = useState(0);

  const activeImage = images[activeIndex];

  return (
    <div className={cn('grid gap-4', className)} {...props}>
      <Image src={activeImage.src} alt={activeImage.alt} />
      <ul className="flex justify-center gap-4 flex-wrap">
        {images.map(({ src, alt }, index) => {
          return (
            <li key={index}>
              <label className="group cursor-pointer">
                <input
                  type="radio"
                  name={`${id}__product_gallery`}
                  checked={activeIndex === index}
                  onChange={() => setActiveIndex(index)}
                  className="sr-only"
                />
                <Image
                  src={src}
                  alt={alt}
                  className="w-14 h-14 group-focus-within:tw-cms-outline"
                />
              </label>
            </li>
          );
        })}
      </ul>
    </div>
  );
};

export { ProductGallery };
