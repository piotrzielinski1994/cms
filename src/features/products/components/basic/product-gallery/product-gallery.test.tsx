import { render } from '@testing-library/react';
import { ComponentProps } from 'react';
import { describe, expect, it } from 'vitest';
import { axe } from 'vitest-axe';
import { ProductGallery } from './product-gallery';

describe('ProductGallery', () => {
  const defaultProps = {
    images: [1, 2, 3].map((it) => ({
      src: `/image-${it}.webp`,
      alt: `Image ${it}`,
    })),
  } satisfies ComponentProps<typeof ProductGallery>;

  it('should have no accessibility violations', async () => {
    const { container } = render(<ProductGallery {...defaultProps} />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });

  it('should match the snapshot', async () => {
    const { container } = render(<ProductGallery {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });
});
