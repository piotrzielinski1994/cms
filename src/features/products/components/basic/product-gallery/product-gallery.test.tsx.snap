// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProductGallery > should match the snapshot 1`] = `
<div>
  <div
    class="grid gap-4"
  >
    <dialog>
      <img
        alt="Image 1"
        class="w-full h-auto object-cover cms-image"
        data-nimg="1"
        decoding="async"
        height="0"
        loading="lazy"
        sizes="100vw"
        src="/_next/image?url=%2Fimage-1.webp&w=3840&q=75"
        srcset="/_next/image?url=%2Fimage-1.webp&w=640&q=75 640w, /_next/image?url=%2Fimage-1.webp&w=750&q=75 750w, /_next/image?url=%2Fimage-1.webp&w=828&q=75 828w, /_next/image?url=%2Fimage-1.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimage-1.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimage-1.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimage-1.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimage-1.webp&w=3840&q=75 3840w"
        style="color: transparent;"
        width="0"
      />
    </dialog>
    <button
      class="focus-visible:tw-cms-outline"
      type="button"
    >
      <img
        alt="Image 1"
        class="w-full h-auto object-cover cms-image"
        data-nimg="1"
        decoding="async"
        height="0"
        loading="eager"
        sizes="100vw"
        src="/_next/image?url=%2Fimage-1.webp&w=3840&q=75"
        srcset="/_next/image?url=%2Fimage-1.webp&w=640&q=75 640w, /_next/image?url=%2Fimage-1.webp&w=750&q=75 750w, /_next/image?url=%2Fimage-1.webp&w=828&q=75 828w, /_next/image?url=%2Fimage-1.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimage-1.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimage-1.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimage-1.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimage-1.webp&w=3840&q=75 3840w"
        style="color: transparent;"
        width="0"
      />
    </button>
    <ul
      class="flex justify-center gap-4 flex-wrap"
    >
      <li>
        <label
          class="group cursor-pointer"
        >
          <input
            checked=""
            class="sr-only"
            name="«r1»__product_gallery"
            type="radio"
          />
          <img
            alt="Image 1"
            class="object-cover cms-image w-14 h-14 group-has-[:focus-visible]:tw-cms-outline"
            data-nimg="1"
            decoding="async"
            height="0"
            loading="lazy"
            sizes="3.5rem"
            src="/_next/image?url=%2Fimage-1.webp&w=3840&q=75"
            srcset="/_next/image?url=%2Fimage-1.webp&w=16&q=75 16w, /_next/image?url=%2Fimage-1.webp&w=32&q=75 32w, /_next/image?url=%2Fimage-1.webp&w=48&q=75 48w, /_next/image?url=%2Fimage-1.webp&w=64&q=75 64w, /_next/image?url=%2Fimage-1.webp&w=96&q=75 96w, /_next/image?url=%2Fimage-1.webp&w=128&q=75 128w, /_next/image?url=%2Fimage-1.webp&w=256&q=75 256w, /_next/image?url=%2Fimage-1.webp&w=384&q=75 384w, /_next/image?url=%2Fimage-1.webp&w=640&q=75 640w, /_next/image?url=%2Fimage-1.webp&w=750&q=75 750w, /_next/image?url=%2Fimage-1.webp&w=828&q=75 828w, /_next/image?url=%2Fimage-1.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimage-1.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimage-1.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimage-1.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimage-1.webp&w=3840&q=75 3840w"
            style="color: transparent;"
            width="0"
          />
        </label>
      </li>
      <li>
        <label
          class="group cursor-pointer"
        >
          <input
            class="sr-only"
            name="«r1»__product_gallery"
            type="radio"
          />
          <img
            alt="Image 2"
            class="object-cover cms-image w-14 h-14 group-has-[:focus-visible]:tw-cms-outline opacity-80"
            data-nimg="1"
            decoding="async"
            height="0"
            loading="lazy"
            sizes="3.5rem"
            src="/_next/image?url=%2Fimage-2.webp&w=3840&q=75"
            srcset="/_next/image?url=%2Fimage-2.webp&w=16&q=75 16w, /_next/image?url=%2Fimage-2.webp&w=32&q=75 32w, /_next/image?url=%2Fimage-2.webp&w=48&q=75 48w, /_next/image?url=%2Fimage-2.webp&w=64&q=75 64w, /_next/image?url=%2Fimage-2.webp&w=96&q=75 96w, /_next/image?url=%2Fimage-2.webp&w=128&q=75 128w, /_next/image?url=%2Fimage-2.webp&w=256&q=75 256w, /_next/image?url=%2Fimage-2.webp&w=384&q=75 384w, /_next/image?url=%2Fimage-2.webp&w=640&q=75 640w, /_next/image?url=%2Fimage-2.webp&w=750&q=75 750w, /_next/image?url=%2Fimage-2.webp&w=828&q=75 828w, /_next/image?url=%2Fimage-2.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimage-2.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimage-2.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimage-2.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimage-2.webp&w=3840&q=75 3840w"
            style="color: transparent;"
            width="0"
          />
        </label>
      </li>
      <li>
        <label
          class="group cursor-pointer"
        >
          <input
            class="sr-only"
            name="«r1»__product_gallery"
            type="radio"
          />
          <img
            alt="Image 3"
            class="object-cover cms-image w-14 h-14 group-has-[:focus-visible]:tw-cms-outline opacity-80"
            data-nimg="1"
            decoding="async"
            height="0"
            loading="lazy"
            sizes="3.5rem"
            src="/_next/image?url=%2Fimage-3.webp&w=3840&q=75"
            srcset="/_next/image?url=%2Fimage-3.webp&w=16&q=75 16w, /_next/image?url=%2Fimage-3.webp&w=32&q=75 32w, /_next/image?url=%2Fimage-3.webp&w=48&q=75 48w, /_next/image?url=%2Fimage-3.webp&w=64&q=75 64w, /_next/image?url=%2Fimage-3.webp&w=96&q=75 96w, /_next/image?url=%2Fimage-3.webp&w=128&q=75 128w, /_next/image?url=%2Fimage-3.webp&w=256&q=75 256w, /_next/image?url=%2Fimage-3.webp&w=384&q=75 384w, /_next/image?url=%2Fimage-3.webp&w=640&q=75 640w, /_next/image?url=%2Fimage-3.webp&w=750&q=75 750w, /_next/image?url=%2Fimage-3.webp&w=828&q=75 828w, /_next/image?url=%2Fimage-3.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimage-3.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimage-3.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimage-3.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimage-3.webp&w=3840&q=75 3840w"
            style="color: transparent;"
            width="0"
          />
        </label>
      </li>
    </ul>
  </div>
</div>
`;
